_includes/langselect.html

```diff
@@ -1,17 +1,218 @@
+                {% capture langlink %}{{ page.url }}{% endcapture %}
+                {% capture lang_rel %}canonical{% endcapture %}
+              {% else %}
+                {% capture langlink %}/{{ lng }}{{ page.url }}{% endcapture %}
+                {% capture lang_rel %}alternate{% endcapture %}
+              {% endif %}
+              <li><a href="{{ langlink }}" rel="{{ lang_rel }}" data-slang="{{ lng }}">{{ site.language_names[lng] }}</a></li>
+            {% endif %}
+          {% endfor %}
+        </ul>
+      </div>
+    </div>
+  </div>
 </div>
+<script>
+  document.addEventListener('DOMContentLoaded', function() {
+    var dropdownButton = document.querySelector('.dropdown-button');
+    var dropdownContent = document.querySelector('.dropdown-content');
+
+    dropdownButton.addEventListener('click', function() {
+      dropdownContent.classList.toggle('show');
+    });
+
+    window.onclick = function(event) {
+      if (!event.target.matches('.dropdown-button') && !event.target.matches('.dropdown-button *')) {
+        var dropdowns = document.getElementsByClassName("dropdown-content");
+        for (var i = 0; i < dropdowns.length; i++) {
+          var openDropdown = dropdowns[i];
+          if (openDropdown.classList.contains('show')) {
+            openDropdown.classList.remove('show');
+          }
+        }
+      }
+    };
+
+    // Function to normalize language code
+    function normalizeLang(lang) {
+      return lang.toLowerCase().split('-')[0];
+    }
+
+    // Get user's language from browser
+    var userLang = navigator.language || navigator.userLanguage;
+    var normalizedUserLang = normalizeLang(userLang);
+
+    // Site configuration
+    var siteLanguages = {{ site.languages | jsonify }};
+    var activeLang = '{{ site.active_lang }}';
+    var defaultLang = '{{ site.default_lang }}';
+
+    // Get the stored language from session storage
+    var storedLang = sessionStorage.getItem('activeLang');
+
+    // Determine if redirection is needed
+    var shouldRedirect = false;
+    var newLang = '';
+
+    if (!storedLang) {
+      // If no stored language but browser language is different, use it
+      if (siteLanguages.includes(normalizedUserLang) && normalizedUserLang !== activeLang) {
+        shouldRedirect = true;
+        newLang = normalizedUserLang;
+      }
+    } else if (siteLanguages.includes(storedLang) && storedLang !== activeLang) {
+      // If there's a stored language and it's different from the active one, use it
+      shouldRedirect = true;
+      newLang = storedLang;
+    }
+
+    if (shouldRedirect) {
+      var currentUrl = window.location.pathname;
+      var newUrl = newLang === defaultLang ? currentUrl.replace(/^\/[^/]+/, '') : '/' + newLang + currentUrl.replace(/^\/[^/]+/, '');
+
+      // Prevent endless redirect loops by checking if the URL already contains the language
+      if (!window.location.pathname.startsWith('/' + newLang) && window.location.pathname !== newUrl) {
```

Comment:
checking for absolute paths "/" may need some more thought. I'd like to get most of the stuff from the language picker.